docs/specification/draft/client/elicitation.mdx

```diff
@@ -317,16 +521,199 @@ The three response actions are:
+    participant User
+    participant UserAgent as User Agent (Browser)
+    participant 3AS as 3rd Party AS
+    participant 3RS as 3rd Party RS
+    participant Client as MCP Client
+    participant Server as MCP Server
+
+    Client->>Server: tools/call
+    Note over Server: Needs 3rd-party authorization for user
+    Note over Server: Store state (which user this auth flow is for)
+    Server->>Client: ElicitationRequired error<br> (mode: "oob", url: "example.com/authorize?...")
+    Note over Client: Client notes the tools/call request can be retried later
+    Client->>User: Present consent to open URL
+    User->>Client: Provide consent
+    Client->>UserAgent: Open URL
+    Client->>Server: Accept response
+    Client-->>Server: elicitation/track request (optional)
+    UserAgent->>3AS: Load authorize route
+    Note over 3AS,User: User interaction (OAuth flow):<br>User consents to scoped MCP Server access
+    3AS->>UserAgent: redirect to MCP Server's redirect_uri
+    UserAgent->>Server: load redirect_uri page
+    Note over Server: Confirm: redirect_uri belongs to MCP Server
+    Note over Server: Confirm: elicitation matches user session<br>Confirm: user is logged into MCP Server or MCP AS
+    Server->>3AS: Exchange authorization code for  OAuth tokens
+    3AS->>Server: Grants tokens
+    Note over Server: Bind tokens to MCP user identity
+    Server-->>Client: elicitation/track response (optional)
+    Client->>Server: Retry tools/call
+    Note over Server: Retrieve token bound to user identity
+    Server->>3RS: Call 3rd-party API
+```
+
+This pattern maintains clear security boundaries while enabling rich integrations with third-party services that require user authorization.
+
 ## Security Considerations
 
-1. Servers **MUST NOT** request sensitive information through elicitation
+1. Clients **MUST** provide clear indication of which server is requesting information
 2. Clients **SHOULD** implement user approval controls
-3. Both parties **SHOULD** validate elicitation content against the provided schema
-4. Clients **SHOULD** provide clear indication of which server is requesting information
-5. Clients **SHOULD** allow users to reject elicitation requests at any time
-6. Clients **SHOULD** implement rate limiting
-7. Clients **SHOULD** present elicitation requests in a way that makes it clear what information is being requested and why
+3. Clients **SHOULD** allow users to reject elicitation requests at any time
+4. Clients **SHOULD** implement rate limiting
+5. Clients **SHOULD** present elicitation requests in a way that makes it clear what information is being requested and why
+6. Servers **MUST** bind elicitation requests to the user's identity
+
+### Safe URL Handling
+
+Clients implementing out-of-band elicitation **MUST** handle URLs carefully to prevent users from unknowingly clicking malicious links.
+
+1. Servers **MUST NOT** include URLs in any message or schema fields as part of a form elicitation request.
```

Comment:
Servers linking to documentation on how to use parameters is a common case that is pretty benign. Clients just need to present this appropriately to users as they would any other untrusteed URL.

Reply:
I see what you mean. The main thing I want to separate here is: the client MUST implement strict consent checks around the OOB URL.

The question is, do we want to open the door to showing any URLs that do not have that consent check in place?

Reply:
We cannot reasonably prevent it, a server who wants to phish with a URL is in the field is going to phish no matter what the spec says. I think we should just have guidance for clients to display the URL clearly and obtain consent as we do for the OOB URL itself.

Reply:
> a server who wants to phish with a URL is in the field is going to phish no matter what the spec says

This is actually my exact reasoning for being strict here: we cannot rule out a malicious server sending a URL no matter what the spec says. But by being strict here, the _client_ implementations can better defend against it.